util/config/fixpath.py

```diff
@@ -65,16 +70,18 @@ def main(env='PATH', delim=':'):
 
 if __name__ == '__main__':
 
-    if len(argv) < 2:
-        newpath = main()
-    elif len(argv) < 3:
-        newpath = main(env=argv[1])
+
+    parser = optparse.OptionParser(usage=__doc__)
+    parser.add_option('--shell', dest='shell', default='bash',
+                      help='shell being used');
+
+    (options, args) = parser.parse_args()
+
+    if options.shell == 'fish':
+        delim = ' '
     else:
-        if argv[2] == 'fish':
-            # fishy path (space-separated)
-            newpath = main(env=argv[1], delim=' ')
-        else:
-            # Default: bash/sh/csh/tcsh-like path
-            newpath = main(env=argv[1], delim=':')
+        delim = ':'
+
+    newpath = main(args[0], delim=delim)
```

Comment:
Seems a little weird to call main and then print. Normally we just have this part call main or _main directly and then main does everything else (or calls helper routines) (thinking of the chplenv scripts)